src/main/java/io/usethesource/vallang/io/binary/stream/IValueInputStream.java

```diff
@@ -49,6 +50,9 @@ public IValueInputStream(InputStream in, IValueFactory vf, Supplier<TypeStore> t
         while (read < currentHeader.length) {
             read += in.read(currentHeader, read, currentHeader.length - read);
         }
+        if (!Arrays.equals(currentHeader, Header.MAIN)) {
+            throw new IOException("Incorrect file header, are you sure this is a valid file containing a serialized value?");
```

Comment:
pedantic error message (a rhetorical question). "Unexpected file header: "; perhaps print the current header and the expected header inside the message?